Renaming SimplexNoise refs to OpenSimplexNoise

santouits opened the pull request:
Sorry for this useless pr but I think it will prevent headaches in the future if simplex noise is implemented too.

The three arguments against I see in #21569 are:
- It's a long name to write.
- Users are more familiar with simplex noise and don't care.
- It can later be replaced internally with simplex noise.

But,
I don't think it's a long name to write, you just write it once anyway
The name is self explainable, it just adds an open to prefix. Users should care, because it's a different algorithm.
It can't be replaced internally with simplex noise later because it will return different values. Also it doesn't have to be replaced, it can be added.

 @Megalomaniak and @Xrayez if you have any objections to that.
@JFonS 

the code example in the anouncement should be changed too https://godotengine.org/article/simplex-noise-lands-godot-31, along with the change from persistance to persistence

Xrayez commented:
In order to solve all these issues I think it would be more interesting to implement sort of abstract `Noise` class/API that handles the vast amount of noise function types. So `OpenSimplexNoise` could inherit from `Noise` that could be passed to `NoiseTexture` (notice that this name implies that there should be other types of noise, not only `OpenSimplex` one).

That way, `OpenSimplexNoise` wouldn't need to be replaced, and other noise types like simplex noise, worley noise etc. could use the same API without breaking compatibility.

Having said that, I think renaming `SimplexNoise` to `OpenSimplexNoise` would be necessary with the approach described above.

Megalomaniak commented:
> @Megalomaniak and @Xrayez if you have any objections to that.

none, especially since you raise a valid point:
> Users should care, because it's a different algorithm.

Zylann commented:
I'm not sure if inheritance is a good idea, because that would mean tons of avoidable virtual function calls to generate a noise texture.

akien-mga commented on modules/opensimplex/open_simplex_noise.h:
```diff
@@ -1,5 +1,5 @@
 /*************************************************************************/
-/*  simplex_noise.h                                                      */
+/*  open_simplex_noise.h                                                      */
```
The `*/` at the end of the line needs to be put back in the ranks.

akien-mga commented on modules/opensimplex/open_simplex_noise.cpp:
```diff
@@ -1,5 +1,5 @@
 /*************************************************************************/
-/*  simplex_noise.cpp                                                    */
+/*  open_simplex_noise.cpp                                                    */
```
Same here.

santouits committed:
Renaming SimplexNoise refs to OpenSimplexNoise

Because I think github doesn't show history of renamed files
- The original work is done in commit f12a1b8
by JFonS
- Improved Documentation in commit a386af6
- Last change on the files was in commit 463af5b
- and Fixed compiler warnings in commit e5bbcb8bcf8180a54fdcc5c4291b153246d861ce
